### PR TITLE
Add some bool annotations

### PR DESCRIPTION
**Pull request by nickdrozd**
## Steps

- [ ] For new features or bug fixes, add a ChangeLog entry describing what your PR does.
- [ ] Write a good description on what the PR does.

## Type of Changes

|     | Type                   |
| --- | ---------------------- |
| ✓   | :hammer: Refactoring   |

### CODE REVIEW

File: astroid/nodes/scoped_nodes/scoped_nodes.py [hunk 1]
```diff
@@ -1652,7 +1652,9 @@ def is_abstract(self, pass_is_abstract=True, any_raise_is_abstract=False):
         if pass_is_abstract:
             return True
 
-    def is_generator(self):
```
Comment (nickdrozd) [inline astroid/nodes/scoped_nodes/scoped_nodes.py:1655]: `is_abstract` didn't return anything at the very end. Shouldn't Pylint have caught that?

File: astroid/nodes/node_classes.py [hunk 2]
```diff
@@ -947,7 +947,7 @@ def is_argument(self, name):
             return True
         return (
             self.find_argname(name, rec=True)[1] is not None
-            or self.kwonlyargs
+            or bool(self.kwonlyargs)
```
Comment (nickdrozd) [inline astroid/nodes/node_classes.py:950]: Mypy prefers explicit casting.

Comment (Pierre-Sassoulas) [reply astroid/nodes/node_classes.py:950]: We were maybe returning ``self.kwonlyargs`` and not a boolean some of the time (which could be used in a context where it's going to be truthey or falsey, but not necessarily). What do you think @DanielNoord ?

File: astroid/nodes/scoped_nodes/scoped_nodes.py (hunk 1 above)
Comment (Pierre-Sassoulas) [reply astroid/nodes/scoped_nodes/scoped_nodes.py:1655]: None is falsey, so it worked unless we called it as ``is_abstract(...) is False``. But yeah it should have raised an inconsistent return statement (astroid is not at the same level of quality than pylint, some message are disabled).

**Review by Pierre-Sassoulas — COMMENTED** [review]
Wondering if we should deprecate the ``self.kwonlyargs`` before casting it to bool. Probably safe to just cast it.

File: astroid/bases.py [hunk 3]
```diff
@@ -399,7 +399,7 @@ def __repr__(self):
     def implicit_parameters(self) -> Literal[0]:
         return 0
 
-    def is_bound(self):
+    def is_bound(self) -> bool:
```
Comment (DanielNoord) [inline astroid/bases.py:402]: This should be a `Literal` as it can be useful to know this in `if ... else`'s.

File: astroid/bases.py [hunk 4]
```diff
@@ -462,7 +462,7 @@ def _infer_builtin_new(
                 yield Instance(inferred)
             raise InferenceError
 
-    def bool_value(self, context=None):
+    def bool_value(self, context=None) -> bool:
```
Comment (DanielNoord) [inline astroid/bases.py:465]: Preferably a `Literal` as well. Not going to comment about all of them.

File: astroid/nodes/as_string.py [hunk 5]
```diff
@@ -65,7 +65,7 @@ def _precedence_parens(self, node, child, is_left=True):
 
         return child.accept(self)
 
-    def _should_wrap(self, node, child, is_left):
+    def _should_wrap(self, node, child, is_left: bool) -> bool:
```
Comment (DanielNoord) [inline astroid/nodes/as_string.py:68]: How is `is_left` determined? The `!=` check seems weird for a bool.

File: astroid/nodes/node_classes.py (hunk 2 above)
Comment (DanielNoord) [reply astroid/nodes/node_classes.py:950]: I think this is just a by product of the typing not being complete. The bool cast shouldn't be necessary so let's remove for now until we have actual complete typing for `mypy` to work with.

**Review by DanielNoord — CHANGES_REQUESTED** [review]

**Review by Pierre-Sassoulas — APPROVED** [review]
👌

File: astroid/nodes/as_string.py (hunk 5 above)
Comment (nickdrozd) [reply astroid/nodes/as_string.py:71]: It looks like callers just specify it. It's always boolean as far as I can tell.

File: astroid/nodes/scoped_nodes/scoped_nodes.py [hunk 6]
```diff
@@ -1574,7 +1574,7 @@ def igetattr(self, name, context=None):
                 str(error), target=self, attribute=name, context=context
             ) from error
 
-    def is_method(self):
+    def is_method(self) -> bool:
```
Comment (DanielNoord) [inline astroid/nodes/scoped_nodes/scoped_nodes.py:1577]: Could you remove `rtype`?

File: astroid/nodes/scoped_nodes/scoped_nodes.py [hunk 7]
```diff
@@ -1608,7 +1608,7 @@ def decoratornames(self, context=None):
                 continue
         return result
 
-    def is_bound(self):
+    def is_bound(self) -> bool:
```
Comment (DanielNoord) [inline astroid/nodes/scoped_nodes/scoped_nodes.py:1611]: Could you remove `rtype`?

File: astroid/nodes/scoped_nodes/scoped_nodes.py [hunk 8]
```diff
@@ -1652,7 +1652,9 @@ def is_abstract(self, pass_is_abstract=True, any_raise_is_abstract=False):
         if pass_is_abstract:
             return True
 
-    def is_generator(self):
+        return False
+
+    def is_generator(self) -> bool:
```
Comment (DanielNoord) [inline astroid/nodes/scoped_nodes/scoped_nodes.py:1657]: Could you remove `rtype`?

File: astroid/nodes/scoped_nodes/scoped_nodes.py [hunk 9]
```diff
@@ -1741,7 +1743,7 @@ def infer_call_result(self, caller=None, context=None):
                 except InferenceError:
                     yield util.Uninferable
 
-    def bool_value(self, context=None):
+    def bool_value(self, context=None) -> bool:
```
Comment (DanielNoord) [inline astroid/nodes/scoped_nodes/scoped_nodes.py:1746]: Could you remove `rtype`?

File: astroid/nodes/scoped_nodes/scoped_nodes.py [hunk 10]
```diff
@@ -2196,7 +2198,7 @@ def display_type(self) -> str:
         """
         return "Class"
 
-    def callable(self):
+    def callable(self) -> bool:
```
Comment (DanielNoord) [inline astroid/nodes/scoped_nodes/scoped_nodes.py:2201]: Could you remove `rtype`?

File: astroid/nodes/scoped_nodes/scoped_nodes.py [hunk 11]
```diff
@@ -2206,7 +2208,7 @@ def callable(self):
         """
         return True
 
-    def is_subtype_of(self, type_name, context=None):
+    def is_subtype_of(self, type_name, context=None) -> bool:
```
Comment (DanielNoord) [inline astroid/nodes/scoped_nodes/scoped_nodes.py:2211]: Could you remove `rtype`?

File: astroid/nodes/scoped_nodes/scoped_nodes.py [hunk 12]
```diff
@@ -2439,7 +2441,7 @@ def instance_attr_ancestors(self, name, context=None):
             if name in astroid.instance_attrs:
                 yield astroid
 
-    def has_base(self, node):
+    def has_base(self, node) -> bool:
```
Comment (DanielNoord) [inline astroid/nodes/scoped_nodes/scoped_nodes.py:2444]: Could you remove `rtype`?

File: astroid/nodes/scoped_nodes/scoped_nodes.py [hunk 13]
```diff
@@ -2680,7 +2682,7 @@ def igetattr(
                     str(error), target=self, attribute=name, context=context
                 ) from error
 
-    def has_dynamic_getattr(self, context=None):
+    def has_dynamic_getattr(self, context=None) -> bool:
```
Comment (DanielNoord) [inline astroid/nodes/scoped_nodes/scoped_nodes.py:2685]: Could you remove `rtype`?

File: astroid/nodes/scoped_nodes/scoped_nodes.py [hunk 14]
```diff
@@ -3053,7 +3055,7 @@ def mro(self, context=None) -> list[ClassDef]:
         """
         return self._compute_mro(context=context)
 
-    def bool_value(self, context=None):
+    def bool_value(self, context=None) -> Literal[True]:
```
Comment (DanielNoord) [inline astroid/nodes/scoped_nodes/scoped_nodes.py:3058]: Could you remove `rtype`?

File: astroid/util.py [hunk 15]
```diff
@@ -43,7 +43,7 @@ def __getattribute__(self, name):
     def __call__(self, *args, **kwargs):
         return self
 
-    def __bool__(self):
+    def __bool__(self) -> bool:
```
Comment (DanielNoord) [inline astroid/util.py:46]: Literal I think?

**Review by DanielNoord — CHANGES_REQUESTED** [review]
Sorry I only considered the docstring changes after seeing the other PR. This PR already shows the risk of typing and docstring being incompatible with the `Literal` types being used in places where `rtype` is still `bool`.

**Review by DanielNoord — APPROVED** [review]

**Pull request merged**